### ISSUE TITLE
Upgrade generator

### ISSUE THREAD

**Issue by UlisesGascon** [yeoman/generator-chrome-extension#218]
We can upgrade the generator a little bit 💪 💪 

### Some ideas:
- [ ] Migrate to ES6+
- [ ] Reduce support for lower Nodejs versions (I suggest to support from v10 (included))
- [ ] Upgrade dependency `yeoman-generator` to  ^4.2.0. Major code change
- [ ] Upgrade dependency `yeoman-test` to 2.0.0
- [ ] Upgrade dependency `yeoman-assert` to ^3.1.1
- [ ]  Upgrade dependency `generator-jasmine` to ^2.0.1 
- [ ] Upgrade dependency `generator-mocha` to ^2.0.3 

🚀 More ideas are super welcome!. 

This issue is great for newcomers and first contributions, so... don't be shy and let us know if you want to help us 😉

**Comment by KoolTheba** [comment]
Hi folks!
I'd love to help with this issue. I can take care of the ES6 migration and the upgrade dependency for Yeoman generator.

**Comment by UlisesGascon** [comment]
Thanks @KoolTheba! I asigned you the tasks you requested

**Comment by HemantPawar** [comment]
**TLDR:** _Generated extensions will work but Chrome webstore will not approve the generated extensions._

A lot of things have changed in the past few years. 

Chrome webstore now doesn't approve extension with bundler generated/obfuscated code. So we may have to drop babel, gulp, bower, uglify etc.

Also, this generator was built a long time ago when Chrome was on ES5. But now Chrome supports ES6+ and modules. So we can safetly drop some dependencies as well.

**Comment by HemantPawar** [comment]
Here is the related snippet from [Developer Program Policies
](https://developer.chrome.com/webstore/program_policies)

```
Code Readability Requirements:
Developers must not obfuscate code or conceal functionality of their extension. 
This also applies to any external code or resource fetched by the extension package. 
Minification is allowed, including the following forms:

- Removal of whitespace, newlines, code comments, and block delimiters
- Shortening of variable and function names
- Collapsing files together

```
